sdk/storage/azure-storage-perf/src/main/java/com/azure/storage/blob/perf/DownloadBlobTest.java

```diff
@@ -39,18 +42,31 @@ public void run() {
 
     static class NullOutputStream extends OutputStream {
         @Override
-        public void write(int b) throws IOException {
+        public void write(int b) {
 
         }
+
+        @Override
+        public void write(byte[] b) {
+        }
+
+        @Override
+        public void write(byte[] b, int off, int len) {
+        }
     }
 
     @Override
     public Mono<Void> runAsync() {
         return blobAsyncClient.download()
             .map(b -> {
-                for (int i = 0; i < b.remaining(); i++) {
-                    b.get();
+                int readCount = 0;
```

Comment:
Looking at the [docs ](https://docs.microsoft.com/en-us/java/api/com.azure.storage.blob.specialized.blobasyncclientbase.download?view=azure-java-stable#com_azure_storage_blob_specialized_BlobAsyncClientBase_download__) for this API, would it be better to use this which is simpler, more closely matches the doc sample, and also more closely matches the sync code?

```java
NullOutputStream.getInstance().write(b.array())
```

Reply:
Hmm, that sample should be updated as a `ByteBuffer` isn't guaranteed to have a backing `byte[]`. If the `ByteBuffer` instance is a `DirectByteBuffer` (aka OS managed memory) that will throw an exception.